### PR TITLE
Feat: support ssl_verification_mode => 'full' / 'none'

### PR DESCRIPTION
**Pull request by kares**
the feature itself comes from the mixin: https://github.com/logstash-plugins/logstash-mixin-http_client/pull/39

### CODE REVIEW

File: logstash-output-http.gemspec [hunk 1]
```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |s|
 
   # Gem dependencies
   s.add_runtime_dependency "logstash-core-plugin-api", ">= 1.60", "<= 2.99"
-  s.add_runtime_dependency "logstash-mixin-http_client", ">= 6.0.0", "< 8.0.0"
+  s.add_runtime_dependency "logstash-mixin-http_client", ">= 7.1.0", "< 8.0.0"
```
Comment (kares) [inline logstash-output-http.gemspec:23]: 7.1.0 contains support for the `ssl_verification_mode` option

**Comment by kares** [comment]
:red_circle: 8.x seems to be hitting a JNR bug (triggered by WEBrick used during testing):
```
logstash_1_ef554d540a5f | Jan 25, 2022 1:54:43 PM jnr.netdb.NativeProtocolsDB load
logstash_1_ef554d540a5f | WARNING: Failed to load native protocols db
logstash_1_ef554d540a5f | java.lang.RuntimeException: getprotobyname_r failed
logstash_1_ef554d540a5f | 	at org.jruby.dist/jnr.netdb.NativeProtocolsDB$LinuxNativeProtocolsDB.getProtocolByName(NativeProtocolsDB.java:180)
logstash_1_ef554d540a5f | 	at org.jruby.dist/jnr.netdb.NativeProtocolsDB.load(NativeProtocolsDB.java:80)
```

File: docs/index.asciidoc [hunk 2]
```diff
@@ -337,6 +338,18 @@ If encountered as response codes this plugin will retry these requests
 
 Timeout (in seconds) to wait for data on the socket. Default is `10s`
 
+[id="plugins-{type}s-{plugin}-ssl_verification_mode"]
+===== `ssl_verification_mode`
+
+  * Value type is <<string,string>>
+  * Supported values are: `full`, `none`
+  * Default value is `full`
```
Comment (kaisecheng) [inline docs/index.asciidoc:346]: I am thinking what does `full` mean. Beats has a nice [doc](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-ssl.html)
Maybe a good reference.

Comment (kares) [reply docs/index.asciidoc:346]: OK, thought I would keep it short but we could copy the description from Beats' docs ...

**Review by kaisecheng — APPROVED** [review]
code, test and doc look good 🍎

File: docs/index.asciidoc [hunk 3]
```diff
@@ -337,6 +338,22 @@ If encountered as response codes this plugin will retry these requests
...
+
+Controls the verification of server certificates.
+The `full` option verifies that the provided certificate is signed by a trusted authority (CA)
+and also that the server’s hostname (or IP address) matches the names identified within the certificate.
+
+The `none` setting performs no verification of the server’s certificate.
+This mode disables many of the security benefits of SSL/TLS and should only be used after cautious consideration.
+It is primarily intended as a temporary diagnostic mechanism when attempting to resolve TLS errors;
+its use in production environments is strongly discouraged.
```
Comment (karenzone) [inline docs/index.asciidoc:355]:
```suggestion
It is primarily intended as a temporary diagnostic mechanism when attempting to resolve TLS errors.
Using `none`  in production environments is strongly discouraged.
```

Comment (karenzone) [reply docs/index.asciidoc:355]: Minor suggestion for clarity and to simplify construction for translation.

**Review by karenzone — APPROVED** [review]
Left minor suggestion inline for your consideration. Otherwise, LGTM.

**Comment by b2az** [comment]
@kares u my hero

**Pull request merged**